### DISCUSSION TITLE
The local file system returned an error with the following message: Operation not permitted error message

### DISCUSSION THREAD

**Discussion by aolinto** [abraunegg/onedrive#1556] [Q&A] [answered]
Hi,

I'm running `onedrive v2.4.10-0~202102221319~ubuntu20.04.1`

I would like to place OneDrive folder at /mnt/MyData, which is a ntfs partition shared with a Windows installation (that I seldom use).

I copied the config file at found /usr/share/doc/onedrive/ to ~/.config/onedrive/ and modified line: sync_dir = "/mnt/MyData/OneDrive" ('#' removed). Also I indicated in ~/.config/onedrive/sync_list the folder I want to sync (AtWork).

Nevertheless when I try to sync I get:
```
onedrive --synchronize --resync
Configuration file successfully loaded
config file has been updated, checking if --resync needed
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Creating local directory: AtWork

ERROR: The local file system returned an error with the following message:
  Error Message:    AtWork: Operation not permitted
  Calling Function: applyNewItem()

Uploading differences of /mnt/MyData/OneDrive
Uploading new items of /mnt/MyData/OneDrive
```
The folder OneDrive is created with the folder AtWork inside but the files are not transferred.

When I place OneDrive folder at my home folder everything works fine.

Is there any way to solve this problem?

Thanks

Antonio

**Comment by abraunegg** [comment, marked as answer]
@aolinto 
> I'm running `onedrive v2.4.10-0~202102221319~ubuntu20.04.1`

You are running an unsupported client version. Please upgrade your client: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md#installing-from-distribution-packages

Once you have upgraded your client, please provide:
* Application configuration as per `onedrive --display-config`
* Verbose application logging
* Debug application logging

These are needed to work out why the operation is not being permitted .. at the moment it feels like / sounds like a Unix / Linux permission issue to your NTFS location.

**Reply by aolinto** [reply]
@abraunegg 

Now running onedrive v2.4.12-14-g54a6575

To test, I choose one folder to sync ("AtWork") and inside this folder there is only one file ("20210528_122301 - PMAP.jpg").

The target NTFS partition is mounted at "/mnt/Dados". As user I'm in sudo and plugdev groups and I can access this partition without problems. MEGAsync folder is in this partition too. 

Again, the folder was created but the file was not downloaded.

Below follows what I got, thanks for your attention.

```
:~/onedrive$ onedrive --synchronize --resync
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Creating local directory: AtWork

ERROR: The local file system returned an error with the following message:
  Error Message:    AtWork: Operation not permitted
  Calling Function: applyNewItem()

Uploading differences of /mnt/Dados/OneDrive
Uploading new items of /mnt/Dados/OneDrive

:~/onedrive$ onedrive --display-config
Configuration file successfully loaded
onedrive version                       = v2.4.12-14-g54a6575
Config path                            = /home/antonio/.config/onedrive
Config file found in config path       = true
Config option 'check_nosync'           = false
Config option 'sync_dir'               = /mnt/Dados/OneDrive
Config option 'skip_dir'               = 
Config option 'skip_file'              = ~*|.~*|*.tmp
Config option 'skip_dotfiles'          = false
Config option 'skip_symlinks'          = false
Config option 'monitor_interval'       = 300
Config option 'min_notify_changes'     = 5
Config option 'log_dir'                = /var/log/onedrive/
Config option 'classify_as_big_delete' = 1000
Config option 'upload_only'            = false
Config option 'no_remote_delete'       = false
Config option 'remove_source_files'    = false
Config option 'sync_root_files'        = false
Selective sync 'sync_list' configured  = true
sync_list contents:
AtWork
Business Shared Folders configured     = false

:~/onedrive$ systemctl --user stop onedrive
Warning: The unit file, source configuration file or drop-ins of onedrive.service changed on disk. Run 'systemctl --user daemon-reload' to reload units.

:~/onedrive$ systemctl --user daemon-reload

:~/onedrive$ onedrive --synchronize --resync
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Syncing changes from OneDrive ...
Downloading file AtWork/20210528_122301 - PMAP.jpg ... 
Downloading 100% |oooooooooooooooooooooooooooooooooooooooo| DONE IN 00:00:02                                                                                                                                       

ERROR: The local file system returned an error with the following message:
  Error Message:    AtWork/20210528_122301 - PMAP.jpg: Operation not permitted
  Calling Function: downloadFileItem()

Uploading differences of /mnt/Dados/OneDrive
Uploading new items of /mnt/Dados/OneDrive

:~/onedrive$ systemctl --user start onedrive

:/mnt$ ls -l
total 12
drwxrwx--- 1 root plugdev 4096 jul 10 18:16 Dados
```

**Reply by abraunegg** [reply]
@aolinto 
You have a file system permission issue:
```
ERROR: The local file system returned an error with the following message:
  Error Message:    AtWork/20210528_122301 - PMAP.jpg: Operation not permitted
```

The error message is fairly clear on this. This is not a fault or issue with the application.

When running the client under systemd, it is not running it as the user you 'think' it is - and your folder / group ownership will not be right to allow that user to read|write to the location that you want.

Things for you to do:
* Fix your filesystem permissions so that the user running the 'onedrive' process under systemd can access your NTFS filesystem
  * Make sure you are mounting the NTFS filesystem the right way to allow read|write operations
* Potentially look at your SELinux rules - that could be preventing the read|write operations due to how you are mounting your NTFS volume
* Dont use Ubuntu or any Ubuntu variant
  * Test what you want to do on something other that Ubuntu / Ubuntu variant .. It could be your choice of Linux distribution here causing your issue

**Reply by aolinto** [reply]
@abraunegg

I really can not figure out how to overcome this issue. As user I can use this partition to read, write, delete files, etc. When I put a new file into the local folder 'Onedrive / AtWork' it is uploaded. The problem is with the download and writing. I compared the permissions of this folder and, e.g., with the folders of MEGAsync and DropBox and I didn't see any difference. SELinux is not installed in my computer.

As you mentioned, probably the issue is related to the fact that "when running the client under systemd, it is not running it as the user you 'think' it is". I even set umask to 000 in fstab, but the error message was the same.

As my knowledge of Linux is somewhat limited, I couldn't identify which user this would be. But I understand that this is no longer a issue with onedrive application.

Thanks again for your time.
